JASP-Engine/JASP/R/correlation.R

```diff
@@ -14,644 +14,961 @@
+.corrMarginalDistribution <- function(variable, varName, options, xName = NULL, yName = "Density", errors, coord_flip = FALSE){
+  if(isFALSE(options$plotDensities)) return(.displayError(errorMessage = "")) # return empty plot
+  if(length(variable) < 3) return(.displayError(errorMessage = "Plotting not possible:\n Number of observations is < 3"))
+  if(any(is.infinite(variable))) return(.displayError(errorMessage = "Plotting not possible: Infinite value(s)"))
+
+  
+  if(isTRUE(options$plotRanks)) variable <- rank(variable)
+  
+  p <- .plotMarginalCor(variable = variable, xName = xName, yName = yName)
+  
+  if(coord_flip){
+    p <- p + ggplot2::coord_flip() +
+      ggplot2::theme(axis.ticks.y = ggplot2::element_line(), axis.ticks.x = ggplot2::element_blank(), 
+                     axis.text.x = ggplot2::element_blank())
+  }
+  
+  p
+}
 
-    plot <- correlation.plot
-    plot[["convertible"]] <- TRUE
-    plot[["obj"]] <- content[["obj"]]
-    plot[["data"]] <- content[["png"]]
+.corrScatter <- function(xVar, yVar, options, xBreaks = NULL, yBreaks = NULL, xName = NULL, yName = NULL, 
+                         drawAxes = TRUE) {
+  if(length(xVar) <= 1 || length(yVar) <= 1) return(.displayError(errorMessage = "Plotting not possible:\n Number of observations is < 2"))
+  errors <- .hasErrors(data.frame(xVar = xVar, yVar = yVar), message = 'short', 
+                       type = c('infinity'),
+                       all.target = c("xVar", "yVar"), 
+                       exitAnalysisIfErrors = FALSE)
+  if(is.list(errors)) return(.displayError(errorMessage = sprintf("Plotting not possible: %s", errors$message)))
+  
+  if(isTRUE(options$plotRanks)) {
+    xVar <- rank(xVar)
+    yVar <- rank(yVar)
+  }
+  .plotScatter(xVar = xVar, yVar = yVar, xBreaks = xBreaks, yBreaks = yBreaks, xName = xName, yName = yName, 
+               drawAxes = drawAxes)
+}
 
-    correlation.plot <- plot
+.corrHeatmap <- function(jaspResults, options, corrResults, ready){
+  if(isFALSE(options$plotHeatmap)) return()
+  
+  hw <- 30 + 80*length(options$variables)
+  
+  tests <- c(`Pearson's r` = "pearson",
+             `Spearman's rho` = "spearman",
+             `Kendall's tau B` = "kendall")[c(options$pearson, options$spearman, options$kendallsTauB)]
+  
+  if(length(tests) == 0){
+    return()
+  } else if(length(tests) == 1){
+    plot <- createJaspPlot(title = sprintf("%s heatmap", names(tests)), width = hw, height = hw)
+    plot$dependOn(c("variables", "conditioningVariables", "missingValues", "pearson", "spearman", "kendallsTauB", 
+                    "flagSignificant"))
+    plot$position <- 4
+    jaspResults[['heatmaps']] <- plot
+    
+    if(ready) plot$plotObject <- .corrPlotHeatmap(tests, options, corrResults)
+  } else{
+    heatmaps <- createJaspContainer(title = "Heatmaps")
+    heatmaps$dependOn(c("variables", "conditioningVariables", "missingValues", "pearson", "spearman", "kendallsTauB",
+                        "flagSignificant"))
```

Comment:
Why does this have a dependency on "flagSignificant"?

Reply:
because if flagSignificant == TRUE, the heatmap highlights significant correlations...

Reply:
Ah okay, missed that